wordpress-mobile/WordPress-iOS
# My Sites > no sites: show NRV on rotation

Fixes #15019 

This fixes an issue where the no sites view would disappear on rotation.

Details:
The problem was caused by the empty details view being rendered on top of the primary view (which displays the No Results View). Initially, I tried setting the details VC to `nil`. However, the `collapseSecondary` method crashed because there was no detail VC. So instead, when the view is split in the `separateSecondaryFrom` method, the details is just not shown.

To test:
- On an iPhone that switches between single and split view (ex: iPhone 11 Pro Max), login with an account that has no sites.
- Go to the My Site tab.
- Rotate the device.
- Verify the view can be seen in landscape. (The image is hidden in landscape, as it's meant to be.)

To note, this _only_ fixes the NRV "disappearing". You'll notice if you start the app in landscape, the view is incorrectly split, and the NRV appears on the left. This is related to https://github.com/wordpress-mobile/WordPress-iOS/issues/14547#issuecomment-702217917 and will be resolved separately.

![nrv_rotation](https://user-images.githubusercontent.com/1816888/95140291-c54bcf00-072b-11eb-9b87-3dd078eb82b2.gif)

PR submission checklist:
